src/type_conversion/hiwire.c

```diff
@@ -404,53 +408,45 @@ EM_JS(int, hiwire_get_byteLength, (JsRef idobj), {
   return jsobj['byteLength'];
 });
 
-EM_JS(void, hiwire_copy_to_ptr, (JsRef idobj, int ptr), {
+EM_JS(void, hiwire_copy_to_ptr, (JsRef idobj, void* ptr), {
   var jsobj = Module.hiwire.get_value(idobj);
   // clang-format off
   var buffer = (jsobj['buffer'] !== undefined) ? jsobj.buffer : jsobj;
   // clang-format on
   Module.HEAPU8.set(new Uint8Array(buffer), ptr);
 });
 
-EM_JS(int, hiwire_get_dtype, (JsRef idobj), {
-  var jsobj = Module.hiwire.get_value(idobj);
-  switch (jsobj.constructor.name) {
-    case 'Int8Array':
-      dtype = 1; // INT8_TYPE;
-      break;
-    case 'Uint8Array':
-      dtype = 2; // UINT8_TYPE;
-      break;
-    case 'Uint8ClampedArray':
-      dtype = 3; // UINT8CLAMPED_TYPE;
-      break;
-    case 'Int16Array':
-      dtype = 4; // INT16_TYPE;
-      break;
-    case 'Uint16Array':
-      dtype = 5; // UINT16_TYPE;
-      break;
-    case 'Int32Array':
-      dtype = 6; // INT32_TYPE;
-      break;
-    case 'Uint32Array':
-      dtype = 7; // UINT32_TYPE;
-      break;
-    case 'Float32Array':
-      dtype = 8; // FLOAT32_TYPE;
-      break;
-    case 'Float64Array':
-      dtype = 9; // FLOAT64_TYPE;
-      break;
-    case 'ArrayBuffer':
-      dtype = 3;
-      break;
-    default:
-      dtype = 3; // UINT8CLAMPED_TYPE;
-      break;
-  }
-  return dtype;
-});
+EM_JS(void,
+      hiwire_get_dtype,
+      (JsRef idobj, char** format_ptr, Py_ssize_t* size_ptr),
+      {
+        if (!Module.hiwire.dtype_map) {
```

Comment:
Not complaining; just trying to document some reasons. Why is this created lazily instead of at `hiwire_init`? Because most users are not very likely to run into this code path?

Reply:
Two thoughts:
1. don't pay for it if you aren't using it, and
2. keeps the instantiation of the map closer in the code to where it is used.

The second thing is the stronger motivating factor for me though.